libs/libarchfpga/src/physical_types.h

```diff
@@ -450,6 +450,10 @@ struct t_class {
 struct t_class_range {
     int low = 0;
     int high = 0;
+
+    int total() const {
```

Comment:
Can you add a small comment to explain it?e.g. "returns the total number of elements in this range". Also, total_num( ) may be a better name